#!/usr/bin/env python

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import *

PACKAGE = "jetyak_uav_utils"
gen = ParameterGenerator()
# In the frame of the ar tag (x right, y up, z front)
gen.add("pad_center_x", double_t, 0, "x offset from master tag",    0, -20,   20)
gen.add("pad_center_y", double_t, 0, "y offset from master tag",    0, -20,   20)
gen.add("pad_center_z", double_t, 0, "z offset from master tag",    1, -20,   20)

gen.add("kp_x", double_t, 0, "kp_x",    .1, 0,   1)
gen.add("kd_x", double_t, 0, "kd_x",    0, 0,   1)
gen.add("ki_x", double_t, 0, "ki_x",    0, 0,   1)

gen.add("kp_y", double_t, 0, "kp_y",    .1, 0,   1)
gen.add("kd_y", double_t, 0, "kd_y",    0, 0,   1)
gen.add("ki_y", double_t, 0, "ki_y",    0, 0,   1)

gen.add("kp_z", double_t, 0, "kp_z",    .1, 0,   1)
gen.add("kd_z", double_t, 0, "kd_z",    0, 0,   1)
gen.add("ki_z", double_t, 0, "ki_z",    0, 0,   1)

gen.add("kp_w", double_t, 0, "kp_w",    .1, 0,   1)
gen.add("kd_w", double_t, 0, "kd_w",    0, 0,   1)
gen.add("ki_w", double_t, 0, "ki_w",    0, 0,   1)

exit(gen.generate(PACKAGE, "land", "LandConstants"))
